app/models/katello/content_view_version.rb

```diff
@@ -280,6 +280,11 @@ def validate_destroyable!(skip_environment_check = false)
       true
     end
 
+    def all_hosts_editable?(lifecycle_environment)
+      ::Host.in_content_view_environment(self.content_view, lifecycle_environment).count ==
+          ::Host.authorized("view_hosts").in_content_view_environment(self.content_view, lifecycle_environment).count
```

Comment:
it would.  What its trying to figure out is if the user has edit access to all hosts associated with that content view and lifecycle env.  If there are 0 hosts then the user has access to edit all hosts.